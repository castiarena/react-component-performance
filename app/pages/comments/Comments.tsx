import React from 'react';
import Layout from '../../components/Layout';
import ButtonAnchor from '../../components/ButtonAnchor';

const Comments = () => (
    <Layout columns={1} gap={40}>
        <div>
            <h1>Comments</h1>
            <p>
                <ButtonAnchor href="./assets/instructions.pdf" download>Download instructions</ButtonAnchor>
            </p>
        </div>
    </Layout>
);

export default Comments;
