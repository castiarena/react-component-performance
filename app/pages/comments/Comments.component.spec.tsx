import React from 'react';
import 'jest-styled-components';
import { BrowserRouter } from 'react-router-dom';
import Comments from './Comments';
import renderWithTheme from '../../testing-tools/renderWithTheme';

describe('Comments component test suite', () => {
    it('smoke test, snapshot only ', () => {
        const { asFragment } = renderWithTheme(
            <BrowserRouter>
                <Comments />
            </BrowserRouter>,
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
