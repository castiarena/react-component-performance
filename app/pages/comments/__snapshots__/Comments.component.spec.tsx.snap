// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments component test suite smoke test, snapshot only  1`] = `
<DocumentFragment>
  .c0 {
  display: grid;
  grid-template-columns: repeat(1,1fr);
  grid-column-gap: 40px;
  padding: 40px;
}

.c1 {
  border: none;
  padding: 16px 30px;
  border-radius: 24px;
  background-color: #726a95;
  color: #fff;
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1:hover {
  cursor: pointer;
  background-color: #8d83b8;
}

.c1:active,
.c1:focus {
  outline: none;
}

@media (max-width:576px) {
  .c0 {
    grid-template-columns: 1fr;
  }
}

<div
    class="c0"
  >
    <div>
      <h1>
        Comments
      </h1>
      <p>
        <a
          class="c1"
          download=""
          href="./assets/instructions.pdf"
        >
          Download instructions
        </a>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
